### PR TITLE
fix: Repair version checking system for Torch

### LINKED ISSUE #2112: 🐛 [Bug] InvalidVersion error when trying to use torch_tensorrt that was compiled in a Jetson container image
##  Bug Description

```
import torch_tensorrt
```

raises an `InvalidVersion` error when installed using:

```bash
python3 setup.py install --use-cxx11-abi
```

on a Jetson Xavier AGX using the `dustynv/ros:humble-pytorch-l4t-r35.3.1` container base image found [here](https://github.com/dusty-nv/jetson-containers). The issue appears to be caused by the fact that `torch.__version__ == 2.0.0.nv23.05`. Changing the line [here](https://github.com/pytorch/TensorRT/blob/7d1d807731c95ffbaa980a3742e289d4184b7871/py/torch_tensorrt/__init__.py#L98):

```python3
if version.parse(torch.__version__) >= version.parse("2.dev"):
```

to:

```python3
if version.parse(".".join(torch.__version__.split(".")[:3])) >= version.parse("2.dev"):
```

makes the error go away.

## To Reproduce

Start a container using the `dustynv/ros:humble-pytorch-l4t-r35.3.1` base image found [here](https://github.com/dusty-nv/jetson-containers).

```bash
apt update
apt install default-jdk unzip zip

PYTHON_VERSION=$(python --version | sed -n 's/^Python \([0-9]\+\.[0-9]\+\).*$/\1/p')

CUDA_VERSION=$(nvcc --version | sed -n 's/^.*release \([0-9]\+\.[0-9]\+\).*$/\1/p')

CUDNN_MAJOR=$(cat /usr/include/cudnn_version.h | sed -n 's/^.*CUDNN_MAJOR \([0-9]\+\).*$/\1/p')
CUDNN_MINOR=$(cat /usr/include/cudnn_version.h | sed -n 's/^.*CUDNN_MINOR \([0-9]\+\).*$/\1/p')
CUDNN_VERSION=${CUDNN_MAJOR}.${CUDNN_MINOR}

TENSORRT_MAJOR=$(cat /usr/include/aarch64-linux-gnu/NvInferVersion.h | sed -n 's/^.*NV_TENSORRT_MAJOR \([0-9]\+\).*$/\1/p')
TENSORRT_MINOR=$(cat /usr/include/aarch64-linux-gnu/NvInferVersion.h | sed -n 's/^.*NV_TENSORRT_MINOR \([0-9]\+\).*$/\1/p')
TENSORRT_VERSION=${TENSORRT_MAJOR}.${TENSORRT_MINOR}

git clone -b v1.4.0 https://github.com/pytorch/TensorRT.git

cd TensorRT
cp toolchains/jp_workspaces/WORKSPACE.jp50 WORKSPACE
sed -i "s/python[0-9]\+.[0-9]\+/python${PYTHON_VERSION}/g" WORKSPACE
sed -i "s/cuda-[0-9]\+.[0-9]\+/cuda-${CUDA_VERSION}/g" WORKSPACE

BAZEL_VERSION=$(cat .bazelversion)
mkdir bazel
cd bazel
curl -fSsL -O https://github.com/bazelbuild/bazel/releases/download/${BAZEL_VERSION}/bazel-${BAZEL_VERSION}-dist.zip
unzip bazel-${BAZEL_VERSION}-dist.zip
bash ./compile.sh
cp output/bazel /usr/local/bin/

cd ../py
TORCH_TENSORRT_VERSION=$(cat versions.py | sed -n 's/^__version__ = "\(.*\)"/\1/p')
echo "__version__ = \"${TORCH_TENSORRT_VERSION}\"" > versions.py
echo "__cuda_version__ = \"${CUDA_VERSION}\"" >> versions.py
echo "__cudnn_version__ = \"${CUDNN_VERSION}\"" >> versions.py
echo "__tensorrt_version__ = \"${TENSORRT_VERSION}\"" >> versions.py

python3 setup.py install --use-cxx11-abi

cd
python3 -c "import torch_tensorrt; print(torch_tensorrt.__version__)"
```

## Expected behavior

No error.

## Environment

A Jetson Xavier AGX and the container base image found [here](https://github.com/dusty-nv/jetson-containers).

### PR DESCRIPTION
**Pull request by gs-olive**
# Description

- Move _C utilities into `ts` directory
- Address version parsing issue for NV versions of Torch
- Add specialized check for NV Torch versions such as `2.0.0.nv23.05`

Versions such as `2.0.0.nv23.05` are incompatible with the `packaging.version` semantic style, and cause invalid version errors upon parsing. Adapting the solution proposed in #2112, we are able to resolve this issue by removing the specialized portion of the version, leaving only the major, minor, and patch values (`x.y.z`).

Fixes #2112

## Type of change

- Bug fix (non-breaking change which fixes an issue)

# Checklist:

- [ x ] My code follows the style guidelines of this project (You can use the linters)
- [ x ] I have performed a self-review of my own code
- [ x ] I have commented my code, particularly in hard-to-understand areas and hacks
- [ x ] I have made corresponding changes to the documentation
- [ x ] I have added tests to verify my fix or my feature
- [ x ] New and existing unit tests pass locally with my changes
- [ x ] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

File: py/torch_tensorrt/__init__.py [hunk 1]
```diff
@@ -94,7 +94,11 @@ def _find_lib(name, paths):
 
 from torch_tensorrt import fx
 
-if version.parse(torch.__version__) >= version.parse("2.1.dev"):
+if version.parse(
+    torch.__version__
```
Comment (narendasan) [inline py/torch_tensorrt/__init__.py:98]: Make a little macro function in utils that just returns the cleaned torch version

Comment (gs-olive) [reply py/torch_tensorrt/__init__.py:98]: Could this function instead go in `py/torch_tensorrt/_version.py`, to avoid the `from torch_tensorrt import _C` which occurs in the `py/torch_tensorrt/_util.py` file?

Comment (narendasan) [reply py/torch_tensorrt/__init__.py:98]: _version.py gets generated by setup.py so probably not a great choice, but i dont really care where it goes, could go in `__init__.py` or somewhere else

Comment (narendasan) [reply py/torch_tensorrt/__init__.py:98]: Alternatively, we probably should be moving _C dep code into .ts anyway. Half the functions could probably be reimplemented without _C

Comment (narendasan) [reply py/torch_tensorrt/__init__.py]: https://github.com/pytorch/TensorRT/blob/d4c9c06213602c6e8d2ce568154ed287ee67dafd/py/torch_tensorrt/_util.py#L12C5-L12C19 This code just references https://github.com/pytorch/TensorRT/blob/main/core/util/build_info.h

Comment (narendasan) [reply py/torch_tensorrt/__init__.py]: I think lets put it in _util.py with an AI to move the _C code out pre 2.1

Comment (narendasan) [reply py/torch_tensorrt/__init__.py]: @gs-olive add a task to do this in https://github.com/pytorch/TensorRT/issues/2119

Comment (gs-olive) [reply py/torch_tensorrt/__init__.py]: This makes sense, I can add the task. I did some minor refactoring of `_util.py` in this PR since it was a minimal change. Let me know if this change is within the scope of the PR/is favorable.

Comment (gs-olive) [reply py/torch_tensorrt/__init__.py]: Added #2121 and rolled back `_util.py` refactoring changes initially made in 38a2c4cf1694d39d5cf0e17df4db07300d0955fd, since they are out of scope for this PR.

**Review by narendasan — APPROVED** [review]
LGTM
